### DISCUSSION TITLE
Passing strings?

### DISCUSSION THREAD

**Discussion by codex-scribe** [js-fns/date-fns#3760] [Q&A] [answered]
I am working with dates on a web application I am building, and filtering a large group of objects by their dates for jobs which have been scheduled by the company.  `scheduledDate` comes back from the database formatted as a string, and I then run a filter on the array of objects to compare `scheduledDate` to a time I set with the `setHours` method.  The filter condition is 

```isEqual(setHours(day,hour), job.scheduledDate)```

The first argument is a Date object, the second is a string.  I'm getting a lot of errors in my console saying 
```Starting with v2.0.0-beta.1 date-fns doesn't accept strings as date arguments. Please use `parseISO` to parse strings. See: https://github.com/date-fns/date-fns/blob/master/docs/upgradeGuide.md#string-arguments 
Calendar@http://localhost:3000/main.```

So it seems like the format of `scheduledDate` is the problem.  But I checked the date-fns.org website, and the documentation for `isEqual()` says that it accepts Date objects, strings, and numbers as arguments:
![image](https://github.com/date-fns/date-fns/assets/118865356/f56e41e3-fd3a-4fa5-974b-99128557af6d)

So which is it?  If strings are not accepted, why does the site say they are accepted for v.3.6.0?

**Comment by fturmel** [comment, marked as answer]
v3 re-introduced dates as strings, more on the [launch blog post](https://blog.date-fns.org/v3-is-out/). You're probably just using v2 but looking at the v3 docs.

**Reply by codex-scribe** [reply]
Yes, turns out we are on 2.30.  We must have implemented it on the app in the last few weeks before v3.  I would have assumed v3 was out for longer since it's on 3.6
